plugins/plugin-hosted/codenvy-machine-hosted/src/main/java/com/codenvy/swarm/client/AwsEcrAuthResolver.java

```diff
@@ -0,0 +1,128 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.swarm.client;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.ecr.AmazonECRClient;
+import com.amazonaws.services.ecr.model.AmazonECRException;
+import com.amazonaws.services.ecr.model.AuthorizationData;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenRequest;
+import com.amazonaws.services.ecr.model.GetAuthorizationTokenResult;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import org.eclipse.che.plugin.docker.client.DockerRegistryDynamicAuthResolver;
+import org.eclipse.che.plugin.docker.client.dto.AuthConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.eclipse.che.dto.server.DtoFactory.newDto;
+
+/**
+ * This class provides actual auth config for AWS ECR.
+ *
+ * @author Mykola Morhun
+ */
+@Singleton
+public class AwsEcrAuthResolver implements DockerRegistryDynamicAuthResolver {
+    private static final Logger LOG = LoggerFactory.getLogger(DockerRegistryDynamicAuthResolver.class);
+
+    private final AwsInitialAuthConfig awsInitialAuthConfig;
+
+    @Inject
+    public AwsEcrAuthResolver(AwsInitialAuthConfig awsInitialAuthConfig) {
+        this.awsInitialAuthConfig = awsInitialAuthConfig;
+    }
+
+    /**
+     * Retrieves actual auth data for Amazon ECR.
+     * Returns null if no credential configured for specified registry.
+     * Note, that credentials is changed every 12 hours.
+     *
+     * @return actual auth config for specified AWS ECR or null if no credentials configured
```

Comment:
Please avoid duplicate description of method. Differentiate provided explanation or remove one of the descriptions